crates/bevy_app/src/app.rs

```diff
@@ -55,6 +55,10 @@ pub struct App {
     /// Typically, it is not configured manually, but set by one of Bevy's built-in plugins.
     /// See `bevy::winit::WinitPlugin` and [`ScheduleRunnerPlugin`](crate::schedule_runner::ScheduleRunnerPlugin).
     pub runner: Box<dyn Fn(App)>,
+    /// The [render init functions](Self::add_render_init) are responsible for completing
+    /// the initialization of plugins once the `runner` has determined that the system is
+    /// ready to support rendering.
+    pub render_inits: Vec<Box<dyn FnOnce(&mut App)>>,
```

Comment:
On the topic of "should this be included in 0.8": I think that largely comes down to "can we make android work better than it does on main / with #5130". If yes, I think I'm happy to merge either as-is or with RenderPlugin-like changes (see my last comment). If no, I think it would be better to hold off and take our time until we find something that improves on the current state of things.